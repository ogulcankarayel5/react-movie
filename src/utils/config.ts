export const IMAGE_BASE_URL = 'http://image.tmdb.org/t/p/'
// Sizes: w300, w780, w1280, original
export const BACKDROP_SIZE = 'original'

export const BACKDROP_RESPONSIVE_SIZE = 'w500'
// w92, w154, w185, w342, w500, w780, original
export const POSTER_SIZE = 'w780'
