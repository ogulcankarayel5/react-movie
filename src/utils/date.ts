export const getYear = (date: string) => {
  const formattedDate = new Date(date)
  return formattedDate.getFullYear().toString()
}
