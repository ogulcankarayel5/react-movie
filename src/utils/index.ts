export * from 'utils/history'
export * from 'utils/globalStyle'
export * from 'utils/theme'
export * from 'utils/config'
