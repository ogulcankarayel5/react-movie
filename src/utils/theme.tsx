import React, { ReactElement, ReactNode } from 'react'
import { ThemeProvider } from 'styled-components'

export const theme = {
  colors: {
    primary: '#111822',
  },
  fonts: {
    montserratBlack: 'MontserratBlack',
  },
}

interface IThemeProps {
  children: ReactNode
}

export const Theme = ({ children }: IThemeProps): ReactElement => (
  <ThemeProvider theme={theme}>{children}</ThemeProvider>
)
