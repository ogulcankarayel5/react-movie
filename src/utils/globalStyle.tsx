import { createGlobalStyle } from 'styled-components'
import { normalize } from 'styled-normalize'
import { breakPoints } from 'utils/breakpoints'

export const GlobalStyle = createGlobalStyle`
  ${normalize}
  
@font-face {
  font-family:'MontserratBlack';
  src: url('../assets/fonts/Montserrat-Black.ttf') format('truetype');
  font-style: normal; 
}

 *,
 *::before,
 *::after {
   box-sizing: inherit;
}
* {
    font-family: 'MontserratBlack';
    -webkit-font-smoothing: antialiased;
    -moz-osx-font-smoothing: grayscale;
    font-smooth: always;
    margin:0;
    padding:0;

   }

  html {
    font-size: 16px;
    box-sizing: border-box;
    @media (max-width: ${breakPoints.md}) {
      font-size:13px;
  }

  @media (max-width: ${breakPoints.sm}) {
      font-size:11px;
  }

  }

  body {
    background-color: ${(props) => props.theme.colors.primary};
  }

  body,html {
    height:100%;
  }
`
