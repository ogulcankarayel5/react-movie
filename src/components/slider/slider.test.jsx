import React from 'react'
import renderer from 'react-test-renderer'
import Slider from 'components/slider/slider'
import { ThemeProvider } from 'styled-components'
import { theme } from 'utils'
import { render, fireEvent, act, screen } from '@testing-library/react'

const renderWithTheme = (comp) => {
  return <ThemeProvider theme={theme}>{comp}</ThemeProvider>
}

describe('snapshots', () => {
  test('should render correctly the slider', () => {
    const comp = renderWithTheme(
      <Slider>
        <Slider.Slide value={0}>
          <Slider.Caption>
            <Slider.Title text='title' />
            <Slider.Text text='description' />
          </Slider.Caption>

          <Slider.Image src='http://image.tmdb.org/t/p/original/iLLDiO4dbUfFEnRug8DuvFEl1NB.jpg' />
        </Slider.Slide>
      </Slider>
    )
    const tree = renderer.create(comp).toJSON()
    expect(tree).toMatchSnapshot()
  })
})

describe('other', () => {
  test('should render slider with dots', () => {
    const comp = renderWithTheme(
      <Slider>
        <Slider.Slide value={0}>
          <Slider.Image src='http://image.tmdb.org/t/p/original/iLLDiO4dbUfFEnRug8DuvFEl1NB.jpg' />
        </Slider.Slide>
      </Slider>
    )
    const { getByTestId } = render(comp)

    expect(getByTestId('dots')).toBeInTheDocument()
  })

  test('should render slider without dots', () => {
    const comp = renderWithTheme(
      <Slider dots={false}>
        <Slider.Slide value={0}>
          <Slider.Image src='http://image.tmdb.org/t/p/original/iLLDiO4dbUfFEnRug8DuvFEl1NB.jpg' />
        </Slider.Slide>
      </Slider>
    )
    const { queryByTestId } = render(comp)

    expect(queryByTestId('dots')).not.toBeInTheDocument()
  })

  test('initial slider should be active, after click the next slider should be active ', () => {
    const comp = renderWithTheme(
      <Slider dots={true}>
        <Slider.Slide value={0}>
          <Slider.Image src='http://image.tmdb.org/t/p/original/iLLDiO4dbUfFEnRug8DuvFEl1NB.jpg' />
        </Slider.Slide>
        <Slider.Slide value={1}>
          <Slider.Image src='http://image.tmdb.org/t/p/original/iLLDiO4dbUfFEnRug8DuvFEl1NB.jpg' />
        </Slider.Slide>
      </Slider>
    )
    render(comp)

    expect(screen.queryByTestId('dot-0')).toHaveStyle({ background: '#c51d3c' })
    expect(screen.queryByTestId('dot-1')).toHaveStyle({ background: '#646b71' })

    fireEvent.click(screen.queryByTestId('dot-1'))

    expect(screen.queryByTestId('dot-0')).toHaveStyle({ background: '#646b71' })
    expect(screen.queryByTestId('dot-1')).toHaveStyle({ background: '#c51d3c' })
  })

  test('active index should be changed after autoplaytime that passing the component as prop', async () => {
    jest.useFakeTimers()
    const comp = renderWithTheme(
      <Slider autoPlayTime={3000} dots={true}>
        <Slider.Slide value={0}>
          <Slider.Image src='http://image.tmdb.org/t/p/original/iLLDiO4dbUfFEnRug8DuvFEl1NB.jpg' />
        </Slider.Slide>
        <Slider.Slide value={1}>
          <Slider.Image src='http://image.tmdb.org/t/p/original/iLLDiO4dbUfFEnRug8DuvFEl1NB.jpg' />
        </Slider.Slide>
      </Slider>
    )
    render(comp)
    expect(screen.queryByTestId('dot-0')).toHaveStyle({ background: '#c51d3c' })
    expect(screen.queryByTestId('dot-1')).toHaveStyle({ background: '#646b71' })

    act(() => {
      jest.runAllTimers() // trigger setTimeout
    })
    expect(screen.queryByTestId('dot-0')).toHaveStyle({ background: '#646b71' })
    expect(screen.queryByTestId('dot-1')).toHaveStyle({ background: '#c51d3c' })
  })

  test(`slide's margin values should be right`, () => {
    const comp = renderWithTheme(
      <Slider dots={true}>
        <Slider.Slide value={0}>
          <Slider.Image src='http://image.tmdb.org/t/p/original/iLLDiO4dbUfFEnRug8DuvFEl1NB.jpg' />
        </Slider.Slide>
        <Slider.Slide value={1}>
          <Slider.Image src='http://image.tmdb.org/t/p/original/iLLDiO4dbUfFEnRug8DuvFEl1NB.jpg' />
        </Slider.Slide>
        <Slider.Slide value={2}>
          <Slider.Image src='http://image.tmdb.org/t/p/original/iLLDiO4dbUfFEnRug8DuvFEl1NB.jpg' />
        </Slider.Slide>
      </Slider>
    )
    const { getByTestId } = render(comp)

    expect(getByTestId('0-slide')).toHaveStyle(`margin-left:-0%`)

    fireEvent.click(screen.queryByTestId('dot-1'))

    expect(getByTestId('0-slide')).toHaveStyle(`margin-left:-100%`)

    fireEvent.click(screen.queryByTestId('dot-2'))

    expect(getByTestId('0-slide')).toHaveStyle(`margin-left:-200%`)
  })

  test('it should render the all slider components', () => {
    const comp = renderWithTheme(
      <Slider dots={true}>
        <Slider.Slide value={0}>
          <Slider.Caption>
            <Slider.Title text='title' />
            <Slider.Text text='text' />
          </Slider.Caption>
          <Slider.Image src='http://image.tmdb.org/t/p/original/iLLDiO4dbUfFEnRug8DuvFEl1NB.jpg' />
        </Slider.Slide>
      </Slider>
    )

    const { getByTestId, getByText } = render(comp)

    expect(getByText(/title/i)).toBeInTheDocument()
    expect(getByText(/text/i)).toBeInTheDocument()
    expect(getByTestId('img')).toBeInTheDocument()
    expect(getByTestId('caption')).toBeInTheDocument()
  })
})
