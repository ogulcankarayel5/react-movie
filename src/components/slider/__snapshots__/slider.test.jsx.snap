// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`snapshots should render correctly the slider 1`] = `
<div
  className="sc-gtsrHT bNtySs"
>
  <div
    className="sc-dlnjwi jqJMsp"
    data-testid="0-slide"
    style={
      Object {
        "marginLeft": "-0%",
      }
    }
  >
    <div
      className="sc-hKFxyN bhZNsx"
      data-testid="caption"
    >
      <p
        className="sc-bdnxRM ksaGXG sc-pNWdM fQIpJF"
        size="xl"
      >
        title
      </p>
      <p
        className="sc-bdnxRM fmvsy sc-jrsJWt GCejX"
        size="medium"
      >
        description
      </p>
    </div>
    <div
      className="sc-jSFjdj duCRrc"
    >
      <img
        className="sc-eCApnc jEkbAm"
        data-testid="img"
        src="http://image.tmdb.org/t/p/original/iLLDiO4dbUfFEnRug8DuvFEl1NB.jpg"
      />
    </div>
  </div>
  <div
    className="sc-gKAaRy eUZzQG"
    data-testid="dots"
  >
    <div
      className="sc-iCoGMd jyzrzN"
      data-testid="dot-0"
      onClick={[Function]}
    />
  </div>
</div>
`;
